sotodlib/hwp/hwp.py

```diff
@@ -112,16 +115,19 @@ def demod(aman, signal='signal', name='signal_demod',
         aman.move(name + '_wo_lfilt', None)
     if name in aman.keys():
         aman.move(name, None)
-
-    speed = (np.sum(np.diff(aman.hwp_angle) % (2 * np.pi)) /
-             (aman.timestamps[-1] - aman.timestamps[0])) / (2 * np.pi)
-    prelfilt = tod_ops.filters.low_pass_butter4(fc=4 * speed + bpf_width)
+    
+    if bpf_center is None:
+        speed = (np.sum(np.abs((np.diff(aman.hwp_angle) + np.pi)%(2*np.pi) - np.pi)) / 
```

Comment:
np.unwarp can be used to this process
speed = (np.sum(np.abs(np.diff(np.unwrap(aman.hwp_angle)))) /
               (aman.timestamps[-1] - aman.timestamps[0])) / (2 * np.pi)